atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/util/MeasuringUtilsTest.java

```diff
@@ -0,0 +1,202 @@
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Test;
+
+public final class MeasuringUtilsTest {
+    private static final int SIZE_1 = 123;
+    private static final int SIZE_2 = 214;
+    private static final int SIZE_3 = 329;
+    private static final Measurable MEASURABLE_1 = () -> SIZE_1;
+    private static final Measurable MEASURABLE_2 = () -> SIZE_2;
+
+    @Test
+    public void sizeOfEmptyObjectsIsZero() {
+        assertThat(MeasuringUtils.sizeOf(List.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOf(Set.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOf(Map.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOf(ImmutableMultimap.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOfMeasurableLongMap(Map.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOfMeasurableByteMap(Map.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOfByteCollection(List.of())).isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOfPageByRangeRequestMap(Map.of())).isEqualTo(0);
+    }
+
+    @Test
+    public void sizeOfMeasurableToLongMapIsCorrect() {
+        Map<Measurable, Long> toLong = ImmutableMap.<Measurable, Long>builder()
+                .put(MEASURABLE_1, 0L)
+                .put(MEASURABLE_2, 1L)
+                .buildOrThrow();
+
+        assertThat(MeasuringUtils.sizeOfMeasurableLongMap(toLong)).isEqualTo(SIZE_1 + SIZE_2 + 2L * Long.BYTES);
+    }
+
+    @Test
+    public void sizeOfMeasurableToByteArrayIsCorrect() {
+        Map<Measurable, byte[]> toByteArray = ImmutableMap.<Measurable, byte[]>builder()
+                .put(MEASURABLE_1, createBytes(SIZE_3))
+                .put(MEASURABLE_2, createBytes(SIZE_3))
+                .buildOrThrow();
+
+        assertThat(MeasuringUtils.sizeOfMeasurableByteMap(toByteArray)).isEqualTo(SIZE_1 + SIZE_2 + 2L * SIZE_3);
+    }
+
+    @Test
+    public void sizeOfByteArrayCollectionIsCorrect() {
+        assertThat(MeasuringUtils.sizeOfByteCollection(List.of(PtBytes.EMPTY_BYTE_ARRAY)))
+                .isEqualTo(0);
+        assertThat(MeasuringUtils.sizeOfByteCollection(List.of(createBytes(SIZE_1))))
+                .isEqualTo(SIZE_1);
+        assertThat(MeasuringUtils.sizeOfByteCollection(
+                        List.of(createBytes(SIZE_1), createBytes(SIZE_2), createBytes(SIZE_2))))
+                .isEqualTo(SIZE_1 + 2L * SIZE_2);
+    }
+
+    @Test
+    public void sizeOfRowResultOfSetOfLongIsCorrect() {
+        assertThat(MeasuringUtils.sizeOfLongSetRowResult(RowResult.of(createCellWithByteSize(SIZE_1), Set.of())))
+                .isEqualTo(SIZE_1);
+
+        assertThat(MeasuringUtils.sizeOfLongSetRowResult(RowResult.of(createCellWithByteSize(SIZE_2), Set.of(1L))))
+                .isEqualTo(Long.sum(SIZE_2, Long.BYTES));
```

Comment:
Is there a specific reason we used this as opposed to just `+`?

Reply:
Discussed offline - adding ints